hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -371,43 +390,51 @@ SqlResult execute(DropTypePlan plan) {
         return UpdateSqlResultImpl.createUpdateCountResult(0);
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
     SqlResult execute(ShowStatementPlan plan) {
-        Stream<String> rows;
+        Stream<List<?>> rows;
 
         switch (plan.getShowTarget()) {
             case MAPPINGS:
-                rows = catalog.getMappingNames().stream();
+                rows = catalog.getMappingNames().stream().map(Collections::singletonList);
                 break;
             case VIEWS:
-                rows = catalog.getViewNames().stream();
+                rows = catalog.getViewNames().stream().map(Collections::singletonList);
                 break;
             case JOBS:
                 JetServiceBackend jetServiceBackend = nodeEngine.getService(JetServiceBackend.SERVICE_NAME);
-                rows = jetServiceBackend.getJobRepository().getActiveJobNames().stream();
+                rows = jetServiceBackend.getJobRepository().getActiveJobNames().stream().map(Collections::singletonList);
                 break;
             case TYPES:
-                rows = catalog.getTypeNames().stream();
+                rows = catalog.getTypeNames().stream().map(Collections::singletonList);
                 break;
             case DATACONNECTIONS:
                 InternalDataConnectionService service = nodeEngine.getDataConnectionService();
                 DataConnectionServiceImpl dataConnectionService = (DataConnectionServiceImpl) service;
                 rows = DataConnectionResolver
-                        .getAllDataConnectionEntries(dataConnectionService, dataConnectionCatalog.getDataConnectionStorage())
-                        .stream()
-                        .map(DataConnectionCatalogEntry::name);
+                        .getAllDataConnectionNameWithTypes(dataConnectionService)
+                        .stream();
                 break;
             case RESOURCES:
                 return executeShowResources(plan.getDataConnectionName());
             default:
                 throw new AssertionError("Unsupported SHOW statement target");
         }
-        SqlRowMetadata metadata = new SqlRowMetadata(singletonList(new SqlColumnMetadata("name", VARCHAR, false)));
+        SqlRowMetadata metadata =
+                plan.getShowTarget() == ShowStatementTarget.DATACONNECTIONS
+                ?  new SqlRowMetadata(asList(
+                        new SqlColumnMetadata("name", VARCHAR, false),
+                        new SqlColumnMetadata("types", JSON, false)
```

Comment:
Why JSON specifically? JSON is just a wrapper type anyway, its not like its going to provide much utility on clients or is there actually a benefit to it?

Reply:
It will be parseable by the JDBC client

Reply:
this causes #24415, OBJECT TYPE without CONNECTION TYPE does not make much sense